contracts/pool/constant-product/ConstantProductPoolFactory.sol

```diff
@@ -29,4 +37,39 @@ contract ConstantProductPoolFactory is PoolDeployer {
         pool = address(new ConstantProductPool{salt: salt}(_deployData, masterDeployer));
         _registerPool(pool, tokens, salt);
     }
+
+    // @dev tokens MUST be sorted i < j => token[i] < token[j]
+    // @dev tokens.length < 256
+    function getPoolsForTokens(address[] calldata tokens)
```

Comment:
I would prefer keeping the Factory minimal and use a separate Helper contract instead for this calculation.

Reply:
I see your point. What others think about it?

Reply:
OK, have moved to the separate contract

Reply:
@sarangparikh22 What do you think?

Reply:
Yeah looks good to me @Okavango

Reply:
Yeah looks good to me @Okavango